apps/unified_connector/tests/test_source.py

```diff
@@ -0,0 +1,87 @@
+from unittest.mock import patch
+
+from utils.graphene.tests import GraphQLTestCase
+from project.factories import ProjectFactory
+
+from unified_connector.factories import (
+    ConnectorSourceFactory,
+    UnifiedConnectorFactory,
+)
+from unified_connector.models import ConnectorSource
+from unified_connector.tests.mock_datas.rss_feed_mock_data import RSS_FEED_MOCK_DATA, expected_rss_data
+from unified_connector.tests.mock_datas.relief_web_mock_data import (
+    RELIEF_WEB_MOCK_DATA_PAGE_1,
+    RELIEF_WEB_MOCK_DATA_PAGE_2,
+    relief_web_expected_data
+)
+from unified_connector.tests.mock_datas.unhcr_mock_data import (
+    UNHCR_MOCK_DATA_PAGE_1,
+    UNHCR_MOCK_DATA_PAGE_2,
+    unhcr_expected_data
+)
+
+
+class TestUnifiedConnectorResponse(GraphQLTestCase):
```

Comment:
Let's do something like this.

```python
from unittest.mock import patch


from unified_connector.models import ConnectorSource


# FIXME: Define this inside tests/mock_data/store.py
CONNECTOR_SOURCE_MOCK_DATA = {
    ConnectorSource.Source.UNHCR: (
        (UNHCR_MOCK_DATA_PAGE_1_RAW, UNHCR_MOCK_DATA_PAGE_2_RAW), UNHCR_WEB_MOCK_EXCEPTED_LEADS
    ),
    ConnectorSource.Source.RELIEF_WEB: (
        (RELIEF_WEB_MOCK_DATA_PAGE_1_RAW, RELIEF_WEB_MOCK_DATA_PAGE_2_RAW), RELIEF_WEB_MOCK_EXCEPTED_LEADS
    ),
}


class ConnectorSourceResponseMock():
    def __init__(self, source_type):
        self.raw_pages_data, self.expected_data = CONNECTOR_SOURCE_MOCK_DATA[source_type]
        self.page = 0

    def get_content_side_effect(self):
        # NOTE: We aren't checking connector_mock_data length
        # here since it should be configured properly in the raw data.
        # Preventing to go to another page
        return self.raw_pages_data[self.page]


class TestConnectorSourceResponse(GraphQLTestCase):

    @staticmethod
    def _assert_lead_equal_to_expected_data(lead_data, expected_data):
        lead_fields = []
        for data in expected_data:
            fields = [str(key) for key, value in data.items()]
            lead_fields += fields
        raw_data = []
        for data in lead_data:
            lead = dict((key, value) for key, value in data.__dict__.items() if key in set(lead_fields))
            raw_data.append(lead)
        assert raw_data == expected_data

    def _connector_response_check(self, source_type, response_mock):
        mock_data = ConnectorSourceResponseMock(source_type)
        response_mock.side_effect = mock_data.get_content_side_effect
        source = ConnectorSourceFactory.create(unified_connector=self.uc, source=source_type)
        leads_result, count = source.source_fetcher().get_leads(source.params)
        self.assertEqual(count, len(mock_data.expected_data))
        self._assert_lead_equal_to_expected_data(leads_result, mock_data.expected_data)

    # ...
    @patch('unified_connector.sources.unhcr_portal.UNHCRPortal.get_content')
    def test_unhcr_response(self, response_mock):
        return self._connector_response_check(ConnectorSource.Source.UNHCR, response_mock)

    @patch('unified_connector.sources.relief_web.ReliefWeb.get_content')
    def test_reliefweb_response(self, response_mock):
        return self._connector_response_check(ConnectorSource.Source.RELIEF_WEB, response_mock)
```